lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/util/TrinoJsonFactory.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.base.util;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonFactoryBuilder;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.StreamReadConstraints;
+import com.fasterxml.jackson.core.TSFBuilder;
+import com.fasterxml.jackson.core.io.CharacterEscapes;
+import com.fasterxml.jackson.core.io.InputDecorator;
+import com.fasterxml.jackson.core.io.OutputDecorator;
+import org.gaul.modernizer_maven_annotations.SuppressModernizer;
+
+public final class TrinoJsonFactory
+        extends JsonFactory
+{
+    private static final String MUTATING_NOT_ALLOWED_MESSAGE = "TrinoJsonFactory does not allow changing configuration. Use TrinoJsonFactory.builder() instead.";
```

Comment:
What is the benefit of blocking those? 
Feels it should be responsibily of code which created instance of `JsonFactory` to not mutate it. And if the code actually wants to do that - why forbid.

Reply:
seems to come as result of https://github.com/trinodb/trino/pull/17854#discussion_r1232227091

Reply:
I do not see how that thread explains why we need to have a subclass of `JsonFactory` which prevents use of mutators. The thread is about droping public shared instance of `TrinoJsonFactory`. While it was needed to have mutators blocked if we had shared instance of `TrinoJsonFactory` managed by the library code, it is no longer useful if we dropped that I think.

Reply:
> We still have shared instances (`JsonUtil` and `JsonOperators`) and I think that this is dangerous to do.

these should become `private static final` fields. maybe something like https://github.com/trinodb/trino/pull/17950 could help